content/blog/deploy-stateful-mongodb-application-on-kubernetes-clusters-in-hpe-greenlake-for-containers-1.md

```diff
@@ -0,0 +1,547 @@
+
+```markdown
+$ kubectl get all -n pce-mongodb
+NAME                        READY   STATUS    RESTARTS   AGE
+pod/pce-mongodb-replica-0   1/1     Running   0          30s
+pod/pce-mongodb-replica-1   1/1     Running   0          19s
+pod/pce-mongodb-replica-2   1/1     Running   0          11s
+
+NAME                  TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)     AGE
+service/pce-mongodb   ClusterIP   None         <none>        27017/TCP   30s
+
+NAME                                   READY   AGE
+statefulset.apps/pce-mongodb-replica   3/3     30s
+```
+
+## Set up MongoDB Replication Host
+
+After MongoDB application is deployed with all the replica pods in running states, you need to set up MongoDB replication. 
+
+#### 1. Connect to the MongoDB pod *pce-mongodb-replica-0*:
+
+```markdown
+$ kubectl exec -it pce-mongodb-replica-0 -n pce-mongodb -- mongo
+MongoDB shell version v4.2.21
+connecting to: mongodb://127.0.0.1:27017/?compressors=disabled&gssapiServiceName=mongodb
+Implicit session: session { "id" : UUID("5d7b5a72-ff37-4ead-b864-965a551dc966") }
+MongoDB server version: 4.2.21
+Welcome to the MongoDB shell.
+For interactive help, type "help".
+For more comprehensive documentation, see
+  https://docs.mongodb.com/
+Questions? Try the MongoDB Developer Community Forums
+  https://community.mongodb.com
+---
+Enable MongoDB's free cloud-based monitoring service, which will then receive and display
+metrics about your deployment (disk utilization, CPU, operation statistics, etc).
+
+The monitoring data will be available on a MongoDB website with a unique URL accessible to you
+and anyone you share the URL with. MongoDB may use this information to make product
+improvements and to suggest MongoDB products and deployment options to you.
+
+To enable free monitoring, run the following command: db.enableFreeMonitoring()
+To permanently disable this reminder, run the following command: db.disableFreeMonitoring()
+---
+
+> 
+```
+
+#### 2. Initiate the replication:
+
+```markdown
+> rs.initiate()
+{
+  "info2" : "no configuration specified. Using a default configuration for the set",
+  "me" : "pce-mongodb-replica-0:27017",
+  "ok" : 1
+}
+rs0:SECONDARY>
+```
+
+#### 3. Add the pod as the *Primary* server to the replication configuration:
+
+```markdown
+rs0:SECONDARY> var cfg = rs.conf()
+rs0:PRIMARY> cfg.members[0].host="pce-mongodb-replica-0.pce-mongodb:27017"
+pce-mongodb-replica-0.pce-mongodb:27017
+rs0:PRIMARY>
+
+rs0:PRIMARY> rs.reconfig(cfg)
+{
+  "ok" : 1,
+  "$clusterTime" : {
+    "clusterTime" : Timestamp(1658309061, 1),
+    "signature" : {
+      "hash" : BinData(0,"AAAAAAAAAAAAAAAAAAAAAAAAAAA="),
+      "keyId" : NumberLong(0)
+    }
+  },
+  "operationTime" : Timestamp(1658309061, 1)
+}
+rs0:PRIMARY>
+```
+
+#### 4. Add the second MongoDB pod *pce-mongodb-replica-1* to the replication configuration:
```

Comment:
"second" or "Secondary"? or "secondary"?

Reply:
The MongoDB app has been deployed with 3 pods, 'pce-mongodb-replica-[0,1,2]'. The first pod 'pce-mongodb-replica-0' has been added as the Primary server (step 3). Here, the *second* pod, i.e., 'pce-mongodb-replica-1', needs to be added to the configuration. Similarly, the next step, step 5, is to add the *third* pod 'pce-mongodb-replica-2' to the configuration. Since the MongoDB Replica Set has one and only one Primary node, the *second* and the *third* pod will be added as the Secondary nodes.